### DISCUSSION TITLE
Temperature Sensor

### DISCUSSION THREAD

**Discussion by sollmaker** [Rem0o/FanControl.Releases#1422] [Q&A]
Hello,

I have a water cooling system and would like to control the fans of my radiators with Fan Control. I have a temperature sensor that measures the water temperature in the cooling circuit. Like most people, I do not have a temperature sensor input on the motherboard. Is there a way to read the temperature sensor value and provide it to Fan Control for regulation? Maybe with an Arduino?

**Comment by Rem0o** [comment]
if it isn't meant to be interfaced with the motherboard directly, yeah a MCU could work, with a basic serial interface communication, then into a File sensor or via a plugin.

**Reply by Rem0o** [reply]
![image](https://user-images.githubusercontent.com/14118956/210097162-7a298312-632b-4c56-8e8b-4333a330c5ca.png)

**Reply by aphexcloud** [reply]
i have tried to do this via a file intergration and while the serial is writing to the file fancontrol can not open the file while its being writen to by the serial writer it also does not support multiple temps in the file. any possibility of having serial added as a sensor type?

[lentiny.txt](https://github.com/Rem0o/FanControl.Releases/files/10378606/lentiny.txt)

**Reply by Rem0o** [reply]
> can not open the file while its being writen to 

Are you releasing the file handle between every writes?

**Reply by Rem0o** [reply]
Also, you must overwrite the value, not append it.

**Reply by secretformula** [reply]
https://github.com/secretformula/FanControl.UsbTemp

**Comment by 37** [comment]
I'm in the exact same position and was wondering the same re: using an MCU. Thanks for the links & cheers for the guidance! Will give it a crack.
